testpar/t_file.c

```diff
@@ -1165,6 +1173,21 @@ test_fapl_preserve_hints(void)
 
     VRFY((info_used != MPI_INFO_NULL), "H5Pget_fapl_mpio");
 
+    memset(key_used, 0, 20);
+    memset(value_used, 0, 20);
+
+    mpi_ret = MPI_Info_get_nthkey(info_used, 0, key_used);
+    VRFY((mpi_ret == MPI_SUCCESS), "MPI_Info_get_nthkey succeeded");
+
+    mpi_ret = MPI_Info_get(info_used, key_used, 20, value_used, &flag);
+    VRFY((mpi_ret == MPI_SUCCESS), "MPI_Info_get succeeded");
+
+    ret = strcmp(key_used, key);
+    VRFY(ret == 0, "strcmp key");
+
+    ret = strcmp(value_used, value);
+    VRFY(ret == 0, "strcmp value");
```

Comment:
I believe these changes are fine, but there's a possibility this test may fail depending on whether MPI keys come exactly in the order they were set on the Info object. Otherwise, the MPI I/O VFD has code to get any info set by the MPI implementation after an MPI_File_open call and then set that back on the FAPL (https://github.com/HDFGroup/hdf5/blob/develop/src/H5FDmpio.c#L892-L934)

Reply:
It may be best to call MPI_Info_get_nkeys and then loop through all the keys, checking to see if any of them match the hint we set here and then failing if none match

Reply:
I will change that based on MPI_Info_get_nkeys.

Reply:
Done